src/elfpy/markets.py

```diff
@@ -320,40 +432,37 @@ def _open_long(self, trade_details):
         #        use current mint time because this is a fresh
         trade_results = self.pricing_model.calc_out_given_in(
             trade_details.trade_amount,
-            trade_details.share_reserves,
-            trade_details.bond_reserves,
+            trade_details.market.share_reserves,
```

Comment:
Is this right? Should this not be `self.share_reserves`, etc? I'm not sure, just checking that you checked.